### PR TITLE
Add New Model for Granted Substitutions

### PR DESCRIPTION
**Pull request by yoomlam**
Resolves [CASEFLOW-1120](https://vajira.max.gov/browse/CASEFLOW-1120)

### Description
Add new model and schema for Granted Substitutions workflow

- substitution date
- substitute/claimant id
- poa id
- appeal id
- created by

### Acceptance Criteria
- [ ] Code compiles correctly
- [ ] New model for granted subs form data
- [ ] Update schema for new model

### Testing Plan
1. Tests, reviewed for coverage, pass
2. Instantiate a GrantedSubstitution in console and interrogate it

### Code Documentation Updates
- [x] Add or update code comments at the top of the class, module, and/or component.

### Database Changes
*Only for Schema Changes*

* [x] Timestamps (created_at, updated_at) for new tables
* [x] Column comments updated
* [x] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`)
* [x] Verify that `migrate:rollback` works as desired ([`change` supported functions](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method))
* [ ] ~Query profiling performed (eyeball Rails log, check bullet and fasterer output)~
* [x] Appropriate indexes added (especially for foreign keys, polymorphic columns, unique constraints, and Rails scopes)
* [x] DB schema docs updated with `make docs` (after running `make migrate`)
* [x] #appeals-schema notified with summary and link to this PR
* [ ] ~Any non-obvious semantics or logic useful for interpreting database data is documented at [Caseflow Data Model and Dictionary](https://github.com/department-of-veterans-affairs/caseflow/wiki/Caseflow-Data-Model-and-Dictionary)~

### CODE REVIEW

**Comment by yoomlam** [comment]
Based on PR CAVC Remand model & Controller #15351 and `create_table ... references` from https://github.com/department-of-veterans-affairs/caseflow/pull/15302/files.

File: app/models/granted_substitution.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  before_save :establish_appeal_stream
+
+  private
+
+  def establish_appeal_stream
+    self.target_appeal ||= source_appeal.create_stream(:granted_substitution).tap do |target_appeal|
+      AdvanceOnDocketMotion.copy_granted_motions_to_appeal(source_appeal, target_appeal)
+      InitialTasksFactory.new(target_appeal, self).create_root_and_sub_tasks!
+    end
+  end
```
Comment (yoomlam) [inline app/models/granted_substitution.rb:22]: Bonus code for https://vajira.max.gov/browse/CASEFLOW-1119

File: spec/models/granted_substitution_spec.rb [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        substitute_id: substitute&.id,
+        poa_participant_id: poa_participant_id
+      }
+    end
+
+    it "creates the record" do
+      expect { subject }.not_to raise_error
+      params.each_key { |key| expect(subject.send(key)).to eq params[key] }
+
+      expect(subject.target_appeal.docket_number).to eq subject.source_appeal.docket_number
```
Comment (yoomlam) [inline spec/models/granted_substitution_spec.rb:27]: Bonus code for https://vajira.max.gov/browse/CASEFLOW-1119

**Review by n1zyy — APPROVED** [review]
This all looks good! Test coverage looks good, schema makes sense, and there are indexes for how I assume we'll access this. :shipit: 🚀

**Pull request merged**